frontend/src/metabase-lib/Question.ts

```diff
@@ -1286,6 +1298,211 @@ class QuestionInner {
     return hasQueryBeenAltered ? question.markDirty() : question;
   }
 
+  generateQueryDescription(tableMetadata, options = {}) {
```

Comment:
This logic is from `queries/utils/description.js`. It'll probably be moved to CLJC functions soon; it makes a nice proof of concept. (Lots of logic, simple return value, self-contained.)